File: 백준/Silver/2606. 바이러스/바이러스.py

N = int(input())
M = int(input())

graph = {}
for i in range(N+1):
    graph[i] = []

vistied = [0]*(N+1)
for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

def dfs(graph, v, visited):

    visited[v]=1 # 방문 표시
    #print(v, end=" ")

    for i in graph[v]:
        if not visited[i]:
            dfs(graph, i, visited)

    # 전체를 다 돌고나서 방문 표시된 노드 확인
    return visited.count(1) - 1  # 1 표시된 노드 개수 반환에 1번째는 빼기


print(dfs(graph, 1, vistied))

File: 백준/Silver/2606. 바이러스/바이러스.kt
package org.example

import java.util.StringTokenizer

fun main() {
    val br = System.`in`.bufferedReader()
    val computerNum = br.readLine().toInt() // 1~100
    val graph = Array(computerNum + 1) { mutableListOf<Int>() }
    val edgeNum = br.readLine().toInt()
    repeat(edgeNum) {
        val st = StringTokenizer(br.readLine())
        val a = st.nextToken().toInt()
        val b = st.nextToken().toInt()
        graph[a].add(b)
        graph[b].add(a)
    }

    // 1번 컴퓨터를 통해 웜 바이러스에 걸리게 되는 컴퓨터의 수
    val visited = BooleanArray(computerNum + 1)
    println(dfs(graph, 1, visited) - 1)

    br.close()
}

fun dfs(graph: Array<MutableList<Int>>, v: Int, visited: BooleanArray): Int {
    var count = 1
    visited[v] = true

    for (neighbor in graph[v]) {
        if (!visited[neighbor]) {
            count += dfs(graph, neighbor, visited)
        }
    }

    return count
}
